sympy/integrals/transforms.py

```diff
@@ -1186,607 +1172,549 @@ def _laplace_build_rules(t, s):
 
+def _laplace_rule_sdiff(f, t, s, doit=True, **hints):
+    """
+    This function looks for multiplications with polynoimials in `t` as they
+    correspond to differentiation in the frequency domain. For example, if it
+    gets ``(t*f(t), t, s)``, it will compute
+    ``-Derivative(LaplaceTransform(f(t), t, s), s)``.
+    """
 
-def _laplace_apply_rules(f, t, s, doit=True, **hints):
+    if f.is_Mul:
+        pfac = [1]
+        ofac = [1]
+        for fac in Mul.make_args(f):
+            if fac.is_polynomial(t):
+                pfac.append(fac)
+            else:
+                ofac.append(fac)
+        if len(pfac)>1:
+            pex = prod(pfac)
+            pc = Poly(pex, t).all_coeffs()
+            N = len(pc)
+            if N>1:
+                debug('_laplace_apply_rules match:')
+                debug('      f, n: %s, %s'%(f, pfac))
+                debug('      rule: frequency derivative (4.1.6)')
+                oex = prod(ofac)
+                r_, p_, c_ = _laplace_transform(oex, t, s, simplify=False)
+                deri = [r_]
+                d1 = False
+                try:
+                    d1 = -diff(deri[-1], s)
+                except ValueError:
+                    d1 = False
+                if r_.has(LaplaceTransform):
+                    for k in range(N-1):
+                        deri.append((-1)**(k+1)*Derivative(r_, s, k+1))
+                else:
+                    if d1:
+                        deri.append(d1)
+                        for k in range(N-2):
+                            deri.append(-diff(deri[-1], s))
+                if d1:
+                    r = Add(*[ pc[N-n-1]*deri[n] for n in range(N) ])
+                    return (r, p_, c_)
+    return None
+
+def _laplace_expand(f, t, s, doit=True, **hints):
     """
-    Helper function for the class LaplaceTransform.
+    This function tries to expand its argument with successively stronger
+    methods: first it will expand on the top level, then it will expand any
+    multiplications in depth, then it will try all avilable expansion methods,
+    and finally it will try to expand trigonometric functions.
 
-    This function does a Laplace transform based on rules and, after
-    applying the rules, hands the rest over to `_laplace_transform`, which
-    will attempt to integrate.
+    If it can expand, it will then compute the Laplace transform of the
+    expanded term.
+    """
+
+    if f.is_Add:
+        return None
+    r = expand(f, deep=False)
+    if r.is_Add:
+        return _laplace_transform(r, t, s, simplify=False)
+    r = expand_mul(f)
+    if r.is_Add:
+        return _laplace_transform(r, t, s, simplify=False)
+    r = expand(f)
+    if r.is_Add:
+        return _laplace_transform(r, t, s, simplify=False)
+    if not r==f:
+        return _laplace_transform(r, t, s, simplify=False)
+    r = expand(expand_trig(f))
+    if r.is_Add:
+        return _laplace_transform(r, t, s, simplify=False)
+    return None
```

Comment:
I think that it would be better to separate these things under some utilities like `expand_integral_add, expand_integral_mul, ...` for users to try these stuff manually, 
Such that we don't have to maintain some sort of 'simplify recipe'.

Reply:
I'm not sure this has such general utility. How would you change this into something that could be used somewhere else?

Reply:
This will be part of one of the next steps that we called "Make it possible to use only rules or only integration." in the points for the future.

Those are:
 - Make simplify default to False.
 - Improve formulae for trig rules.
 - Reduce time taken to build rules.
 - Make it possible to use only rules or only integration.

Reply:
> Such that we don't have to maintain some sort of 'simplify recipe'.

By the way, that simplify recipe is part of what makes the present algorithm particularly powerful ;)

Reply:
I mean some general rewriters like `Integral(Add(...), ...)` -> `Add(Integral(...), ...)`

Reply:
I think that `Integral(f(x) + g(x), x).expand()` works, but doesn't work with laplace transform though

Reply:
I guess `IntegralTransform` could have an `expand` method to distribute sums and pull out coefficients.